server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java

```diff
@@ -304,6 +310,149 @@ static ClusterState addIndexClosedBlocks(final Index[] indices, final Map<Index,
         return ClusterState.builder(currentState).blocks(blocks).metadata(metadata).routingTable(routingTable.build()).build();
     }
 
+    static ClusterState addIndexBlock(final Index[] indices, final Map<Index, ClusterBlock> blockedIndices,
+                                      final ClusterState currentState, APIBlock block) {
+        final Metadata.Builder metadata = Metadata.builder(currentState.metadata());
+
+        final Set<Index> indicesToAddBlock = new HashSet<>();
+        for (Index index : indices) {
+            metadata.getSafe(index); // to check if index exists
+            if (currentState.blocks().hasIndexBlock(index.getName(), block.block)) {
+                logger.debug("index {} already has block {}, ignoring", index, block.block);
+            } else {
+                indicesToAddBlock.add(index);
+            }
+        }
+
+        if (indicesToAddBlock.isEmpty()) {
+            return currentState;
+        }
+
+        final ClusterBlocks.Builder blocks = ClusterBlocks.builder().blocks(currentState.blocks());
+        final RoutingTable.Builder routingTable = RoutingTable.builder(currentState.routingTable());
+
+        for (Index index : indicesToAddBlock) {
+            ClusterBlock indexBlock = null;
+            final Set<ClusterBlock> clusterBlocks = currentState.blocks().indices().get(index.getName());
+            if (clusterBlocks != null) {
+                for (ClusterBlock clusterBlock : clusterBlocks) {
+                    if (clusterBlock.id() == block.block.id()) {
+                        // Reuse the existing UUID-based block
+                        indexBlock = clusterBlock;
+                        break;
+                    }
+                }
+            }
+            if (indexBlock == null) {
+                // Create a new UUID-based block
+                indexBlock = createUUIDBasedBlock(block.block);
+            }
+            assert Strings.hasLength(indexBlock.uuid()) : "Block should have a UUID";
+            blocks.addIndexBlock(index.getName(), indexBlock);
+            blockedIndices.put(index, indexBlock);
+            // update index settings as well to match the block
+            final IndexMetadata indexMetadata = metadata.getSafe(index);
+            if (block.setting().get(indexMetadata.getSettings()) == false) {
+                final Settings updatedSettings = Settings.builder()
+                    .put(indexMetadata.getSettings()).put(block.settingName(), true).build();
+
+                metadata.put(IndexMetadata.builder(indexMetadata)
+                    .settings(updatedSettings)
+                    .settingsVersion(indexMetadata.getSettingsVersion() + 1));
+            }
+        }
+
+        logger.info(() -> new ParameterizedMessage("adding block {} to indices {}", block.name,
+            blockedIndices.keySet().stream().map(Object::toString).collect(Collectors.joining(","))));
```

Comment:
Why use a lambda and parameterized message here? Usually it's used when also passing an exception we want to preserve as a `cause`. Am I missing something?

Reply:
No reason, code was copy-pasted from elsewhere